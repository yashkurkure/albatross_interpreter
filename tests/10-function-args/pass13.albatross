var s string := "";

fun f1 int (a1 int) {
  a1 := a1;
  return a1;
}

fun f2 int (a1 int, a2 string) {
  a1 := a1;
  a2 := s;
  return a1;
}

fun f3 int (a1 int, a2 string, a3 int, a4 string) {
  a1 := a1;
  a2 := s;
  a3 := a1 + a3;
  a4 := a2;
  return a3;
}

fun f4 int (a1 int, a2 string, a3 int, a4 string, a5 int) {
  a1 := a1;
  a2 := s;
  a3 := a1 + a3;
  a4 := a2;
  a5 := a3 + a5;
  return a5;
}

fun f5 int (a1 int, a2 string, a3 int, a4 string, a5 int, a6 string) {
  a1 := a1;
  a2 := s;
  a3 := a1 + a3;
  a4 := a2;
  a5 := a3 + a5;
  a6 := a4;
  return a5;
}

fun f6 int (a1 int, a2 string, a3 int, a4 string, a5 int, a6 string, a7 int) {
  a1 := a1;
  a2 := s;
  a3 := a1 + a3;
  a4 := a2;
  a5 := a3 + a5;
  a6 := a4;
  a7 := a5 + a7;
  return a7;
}

fun f7 int (a1 int, a2 string, a3 int, a4 string, a5 int, a6 string, a7 int, a8 string) {
  a1 := a1;
  a2 := s;
  a3 := a1 + a3;
  a4 := a2;
  a5 := a3 + a5;
  a6 := a4;
  a7 := a5 + a7;
  a8 := s;
  return a7;
}

return f7(400, "CS", 70, "UIC", 4, "Chicago", 2022, "Fall");
