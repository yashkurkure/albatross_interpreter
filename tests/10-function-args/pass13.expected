Variable declared "s" type string
Function declared "f1" returns int
	Argument "a1" type int
Argument/local read "a1" type int
Argument/local written "a1" type int
Argument/local read "a1" type int
Function declared "f2" returns int
	Argument "a1" type int
	Argument "a2" type string
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Function declared "f3" returns int
	Argument "a1" type int
	Argument "a2" type string
	Argument "a3" type int
	Argument "a4" type string
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Argument/local read "a3" type int
Argument/local written "a3" type int
Argument/local read "a2" type string
Argument/local written "a4" type string
Argument/local read "a3" type int
Function declared "f4" returns int
	Argument "a1" type int
	Argument "a2" type string
	Argument "a3" type int
	Argument "a4" type string
	Argument "a5" type int
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Argument/local read "a3" type int
Argument/local written "a3" type int
Argument/local read "a2" type string
Argument/local written "a4" type string
Argument/local read "a3" type int
Argument/local read "a5" type int
Argument/local written "a5" type int
Argument/local read "a5" type int
Function declared "f5" returns int
	Argument "a1" type int
	Argument "a2" type string
	Argument "a3" type int
	Argument "a4" type string
	Argument "a5" type int
	Argument "a6" type string
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Argument/local read "a3" type int
Argument/local written "a3" type int
Argument/local read "a2" type string
Argument/local written "a4" type string
Argument/local read "a3" type int
Argument/local read "a5" type int
Argument/local written "a5" type int
Argument/local read "a4" type string
Argument/local written "a6" type string
Argument/local read "a5" type int
Function declared "f6" returns int
	Argument "a1" type int
	Argument "a2" type string
	Argument "a3" type int
	Argument "a4" type string
	Argument "a5" type int
	Argument "a6" type string
	Argument "a7" type int
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Argument/local read "a3" type int
Argument/local written "a3" type int
Argument/local read "a2" type string
Argument/local written "a4" type string
Argument/local read "a3" type int
Argument/local read "a5" type int
Argument/local written "a5" type int
Argument/local read "a4" type string
Argument/local written "a6" type string
Argument/local read "a5" type int
Argument/local read "a7" type int
Argument/local written "a7" type int
Argument/local read "a7" type int
Function declared "f7" returns int
	Argument "a1" type int
	Argument "a2" type string
	Argument "a3" type int
	Argument "a4" type string
	Argument "a5" type int
	Argument "a6" type string
	Argument "a7" type int
	Argument "a8" type string
Argument/local read "a1" type int
Argument/local written "a1" type int
Variable read "s" type string
Argument/local written "a2" type string
Argument/local read "a1" type int
Argument/local read "a3" type int
Argument/local written "a3" type int
Argument/local read "a2" type string
Argument/local written "a4" type string
Argument/local read "a3" type int
Argument/local read "a5" type int
Argument/local written "a5" type int
Argument/local read "a4" type string
Argument/local written "a6" type string
Argument/local read "a5" type int
Argument/local read "a7" type int
Argument/local written "a7" type int
Variable read "s" type string
Argument/local written "a8" type string
Argument/local read "a7" type int
Function called "f7" returns int
